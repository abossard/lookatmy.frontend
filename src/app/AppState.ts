import {observable} from "mobx";

class AppState {
    @observable public timer = 0;

    constructor() {
        if (process.env.PLATFORM === "browser") {
            setInterval(() => {
                this.timer += 16;
            }, 16);
        }
    }

    public resetTimer() {
        this.timer += 1;
    }
}

export default AppState;
