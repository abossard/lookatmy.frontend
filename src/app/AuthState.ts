import * as ADAL from "adal-ts";
import {User} from "adal-ts";
import {observable} from "mobx";

const config = new ADAL.AdalConfig(
    __AD_CLIENT_ID__,
    __AD_TENANT__,
    __AD_CALLBACK_URL__,
    __AD_CALLBACK_URL__,
    "token",
    `resource=${__AD_RESOURCE__}`,
);

const context = ADAL.Authentication.getContext(config);

class AuthState {
    @observable public loggedIn = false;
    @observable public token?: string;
    @observable public user?: User;

    public login() {
        context.login();
    }

    public process() {
        ADAL.Authentication.getAadRedirectProcessor().process();
        this.token = context.getToken();
        this.user = context.getUser();
        this.loggedIn = this.user !== null && this.token !== null;
    }

    public logout() {
        context.logout();
    }
}

export default AuthState;
