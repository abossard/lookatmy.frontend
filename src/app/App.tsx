import {observer} from "mobx-react";
import DevTools from "mobx-react-devtools";
import * as React from "react";
import {style} from "typestyle";
import AppState from "./AppState";
import AuthState from "./AuthState";

const mainStyle = style({
    color: "green",
    fontFamily: "comic sans, helvetica",
});

const makeSomeSpacePlease = style({
    paddingTop: 20,
});

@observer
class App extends React.Component<{ appState: AppState, authState: AuthState }, {}> {
    public render() {
        return (
            <div className={mainStyle}>
                <h1>look at my <em>xyz</em></h1>
                <hr/>
                <dl>
                    <dt className={makeSomeSpacePlease}>SPA Login, yeah</dt>
                    <dd>
                        {this.props.authState.loggedIn ?
                            <div>Yes, you did it {this.props.authState.user && this.props.authState.user.name}
                                <br/>
                                Your magic cookie token is:
                                <textarea name="token" id="" cols={30} rows={1}>
                                    {this.props.authState.token}
                                </textarea>
                                <br/>
                                <button onClick={this.props.authState.logout}>Logout</button>
                            </div>
                            : <button onClick={this.props.authState.login}>Login</button>
                        }

                    </dd>
                    <dt className={makeSomeSpacePlease}><h3>Async state example</h3></dt>
                    <dd>
                        <p>
                            This button receives the next second from <em>outside</em>. A click on it, resets it to 0.
                        </p>
                        <button onClick={this.onReset}>
                            time passed [ms]: {this.props.appState.timer}
                        </button>
                    </dd>
                </dl>
                <DevTools/>
            </div>
        );
    }

    public onReset = () => {
        this.props.appState.resetTimer();
    }
}

export {App};
