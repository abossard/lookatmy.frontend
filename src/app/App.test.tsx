import * as React from "react";
import {create} from "react-test-renderer";
import {App} from "./App";
import AppState from "./AppState";
import AuthState from "./AuthState";

test("Does render", () => {
    const component = create(
        <App appState={new AppState()} authState={new AuthState()}/>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
