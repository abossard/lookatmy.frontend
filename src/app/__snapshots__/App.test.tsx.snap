// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does render 1`] = `
<div
  className="f8xl9ee"
>
  <h1>
    look at my 
    <em>
      xyz
    </em>
  </h1>
  <hr />
  <dl>
    <dt
      className="f1h5lfj6"
    >
      <h3>
        Login test links
      </h3>
    </dt>
    <dd>
      <a
        href="/auth/facebook"
      >
        Login/Sign Up with Facebook
      </a>
      <br />
      <a
        href="/auth/ad"
      >
        Login/Sign Up with AD
      </a>
      <br />
      <a
        href="/nice"
      >
        Nice site to check if I'm logged in
      </a>
      <br />
      <a
        href="/logout"
      >
        Logout
      </a>
      <br />
    </dd>
    <dt
      className="f1h5lfj6"
    >
      SPA Login, yeah
    </dt>
    <dd>
      <button
        onClick={[Function]}
      >
        Login
      </button>
    </dd>
    <dt
      className="f1h5lfj6"
    >
      <h3>
        Async state example
      </h3>
    </dt>
    <dd>
      <p>
        This button receives the next second from 
        <em>
          outside
        </em>
        . A click on it, resets it to 0.
      </p>
      <button
        onClick={[Function]}
      >
        time passed [ms]: 
        0
      </button>
    </dd>
  </dl>
  <div>
    <div>
      <div
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "rgba(0, 0, 0, 0.1)",
            "borderRadius": "0 0 2px 2px",
            "borderStyle": "solid",
            "borderWidth": "0 1px 1px",
            "color": "rgba(0, 0, 0, 0.8)",
            "display": "flex",
            "fontFamily": "Helvetica, sans-serif",
            "height": "26px",
            "padding": "0 5px",
            "position": "fixed",
            "right": "20px",
            "top": "0px",
            "zIndex": "65000",
          }
        }
      >
        <button
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "active": Object {
                ":hover": Object {
                  "opacity": 1,
                },
                "opacity": 1,
              },
              "background": "transparent none center / 16px 16px no-repeat",
              "backgroundImage": "url(data:image/svg+xml;base64,PHN2ZyBiYXNlUHJvZmlsZT0iYmFzaWMiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmlld0JveD0iMCAwIDE2IDE2IiBoZWlnaHQ9IjE2IiB3aWR0aD0iMTYiPgogICAgPGcgc3Ryb2tlPSIjMDAwIiBmaWxsPSJub25lIiBzdHJva2Utd2lkdGg9IjEiPgogICAgICAgIDxjaXJjbGUgY3g9IjguNSIgY3k9IjguNSIgcj0iNiIvPgogICAgICAgIDxwYXRoIGQ9Ik04LjUgMTBWNCIvPgogICAgPC9nPgogICAgPGcgc3Ryb2tlPSJub25lIiBmaWxsPSIjMDAwIj4KICAgICAgICA8Y2lyY2xlIGN4PSI4LjUiIGN5PSI4LjUiIHI9IjEiLz4KICAgICAgICA8cGF0aCBkPSJNNy41IDFoMnYxLjVoLTJ6Ii8+CiAgICAgICAgPHBhdGggZD0iTTE0IDEuNmwtLjcuNy4zLjMtLjcuOC43LjcuOC0uNy4zLjMuNy0uN3oiLz4KICAgIDwvZz4KPC9zdmc+Cg==)",
              "border": "none",
              "cursor": "pointer",
              "hover": Object {
                "opacity": 0.7,
              },
              "margin": "0 10px",
              "opacity": 0.45,
              "width": "26px",
            }
          }
          title="Visualize component re-renders"
          type="button"
        />
        <button
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "active": Object {
                ":hover": Object {
                  "opacity": 1,
                },
                "opacity": 1,
              },
              "background": "transparent none center / 16px 16px no-repeat",
              "backgroundImage": "url(data:image/svg+xml;base64,PHN2ZyBiYXNlUHJvZmlsZT0iYmFzaWMiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmlld0JveD0iMCAwIDE2IDE2IiBoZWlnaHQ9IjE2IiB3aWR0aD0iMTYiPgogICAgPGcgc3Ryb2tlPSIjMDAwIiBmaWxsPSJub25lIiBzdHJva2Utd2lkdGg9IjEiPgogICAgICAgIDxwYXRoIGQ9Ik0yLjUgMi41aDl2MmgtOXoiLz4KICAgICAgICA8cGF0aCBkPSJNNy41IDcuNWg3djJoLTd6Ii8+CiAgICAgICAgPHBhdGggZD0iTTcuNSAxMi41aDd2MmgtN3oiLz4KICAgICAgICA8cGF0aCBkPSJNNC41IDQuNXY5aDMiLz4KICAgICAgICA8cGF0aCBkPSJNNy41IDguNWgtMyIvPgogICAgPC9nPgo8L3N2Zz4K)",
              "border": "none",
              "cursor": "pointer",
              "hover": Object {
                "opacity": 0.7,
              },
              "margin": "0 10px",
              "opacity": 0.45,
              "width": "26px",
            }
          }
          title="Select a component and show its dependency tree"
          type="button"
        />
        <button
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "active": Object {
                ":hover": Object {
                  "opacity": 1,
                },
                "opacity": 1,
              },
              "background": "transparent none center / 16px 16px no-repeat",
              "backgroundImage": "url(data:image/svg+xml;base64,PHN2ZyBiYXNlUHJvZmlsZT0iYmFzaWMiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmlld0JveD0iMCAwIDE2IDE2IiBoZWlnaHQ9IjE2IiB3aWR0aD0iMTYiPgogICAgPGcgc3Ryb2tlPSIjMDAwIiBmaWxsPSJub25lIiBzdHJva2Utd2lkdGg9IjEiPgogICAgICAgIDxwYXRoIGQ9Ik0xMi41IDMuNWgtOGMtMS4xIDAtMiAuOS0yIDJ2NWMwIDEuMS45IDIgMiAyaDF2Mmw1LTJoMmMxLjEgMCAyLS45IDItMnYtNWMwLTEuMS0uOS0yLTItMnoiLz4KICAgICAgICA8cGF0aCBkPSJNNSA2LjVoNyIvPgogICAgICAgIDxwYXRoIGQ9Ik01IDkuNWg3Ii8+CiAgICA8L2c+Cjwvc3ZnPgo=)",
              "border": "none",
              "cursor": "pointer",
              "hover": Object {
                "opacity": 0.7,
              },
              "margin": "0 10px",
              "opacity": 0.45,
              "width": "26px",
            }
          }
          title="Log all MobX state changes and reactions to the browser console (use F12 to show / hide the console). Use Chrome / Chromium for an optimal experience"
          type="button"
        />
      </div>
    </div>
    <div />
  </div>
</div>
`;
